omegaconf/omegaconf.py

```diff
@@ -715,17 +715,17 @@ def select(
                     throw_on_missing=throw_on_missing,
                     throw_on_resolution_failure=throw_on_resolution_failure,
                 )
-            except (ConfigKeyError, InterpolationResolutionError):
+            except (ConfigKeyError, InterpolationKeyError):
```

Comment:
This is inconsistent with the new behavior of get().
Maybe we should allow InterpolationKeyError to be raised?

Reply:
> This is inconsistent with the new behavior of get().

Can you elaborate on what this new behavior is?

Reply:
Ah, got it, you're right! I had a look to see if it would be easy to have `select()` re-use the same code as `get()`, but it's not entirely obvious, so for now I think what you suggest is what makes most sense.

Done in 8e5d774